src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
This is not what I was expecting.

Implementations such as [in diactoros](https://github.com/zendframework/zend-diactoros/blob/master/src/Stream.php#L299) are essentially a *resource stream*, and the initialization is either from an existing `resource`, or a `string` *path* used to *open* a resource for you.

Having this factory create a *memory* stream or something, and inject the given string as *content* seems inconsistent to me: a string path and a `resource` instance both identify an existing resource, to those seem consistent.

As opposed to providing a chunk of content does not *identify* an *existing* resource, but rather you could say it *is* a resource, and, per the [implementation in guzzle](https://github.com/guzzle/streams/blob/master/src/Stream.php#L49) this actually *creates* a  `php://temp` resource, and writes to it, so that seems inconsistent.

I would have expected one factory that *opens* a resource, whether that resource is a pre-existing `resource` instance or a file-name identifying the resource.

And a separate factory that *creates* a temporary resources and *populates* it for you.

(the latter is how it's implemented in diactoros.)